File: analyze_slides.js

const fs = require('fs');

try {
    const content = fs.readFileSync('presentation.html', 'utf8');

    // Find all slide divs
    // We'll use a regex to split or find matches. 
    // Since we need to process them, let's find indices of <div class="slide

    // Regex for finding slide start
    const slideRegex = /<div class="slide/g;
    let match;
    const slideStarts = [];

    while ((match = slideRegex.exec(content)) !== null) {
        slideStarts.push(match.index);
    }

    const slides = [];
    for (let i = 0; i < slideStarts.length; i++) {
        const start = slideStarts[i];
        const end = (i + 1 < slideStarts.length) ? slideStarts[i + 1] : content.length;
        slides.push(content.substring(start, end));
    }

    console.log("# Presentation Analysis Report");
    console.log(`**File:** \`presentation.html\``);
    console.log(`**Total Slides:** ${slides.length}`);
    console.log("\n## Slide Overview");
    console.log("| Index | Slide Number | Title |");
    console.log("| :--- | :--- | :--- |");

    slides.forEach((slideContent, i) => {
        let title = "No Title";
        let slideNum = "-";

        // Extract H1
        const h1Match = slideContent.match(/<h1[^>]*>([\s\S]*?)<\/h1>/i);
        if (h1Match) {
            let rawTitle = h1Match[1];
            // Remove tags
            title = rawTitle.replace(/<[^>]+>/g, '').trim();

            // Extract number from title
            const numMatch = title.match(/^(\d+)\./);
            if (numMatch) {
                slideNum = numMatch[1];
            }
        }

        // Fallback title check (if h1 didn't work or for title slide)
        if (title === "No Title") {
            // Maybe it's the title slide
            if (slideContent.includes('title-slide')) {
                const heroH1 = slideContent.match(/<h1>([\s\S]*?)<br>/i); // specialized for title slide
                if (heroH1) {
                    title = heroH1[1].replace(/<[^>]+>/g, '').trim();
                }
            }
        }

        // Extract Page Number from footer
        // <div class="footer">...<span>Page 01</span></div>
        const footerMatch = slideContent.match(/class="footer"[^>]*>([\s\S]*?)<\/div>/i);
        if (footerMatch) {
            const footerContent = footerMatch[1];
            const spanMatches = [...footerContent.matchAll(/<span>([\s\S]*?)<\/span>/g)];
            if (spanMatches.length > 0) {
                const lastSpan = spanMatches[spanMatches.length - 1][1];
                if (lastSpan.includes('Page')) {
                    slideNum = lastSpan.replace('Page', '').trim();
                } else if (!isNaN(parseInt(lastSpan.trim()))) {
                    slideNum = lastSpan.trim();
                }
            }
        }

        // Clean title
        title = title.replace(/\n/g, ' ').replace(/\s+/g, ' ');
        if (title.length > 60) {
            title = title.substring(0, 57) + "...";
        }

        console.log(`| ${i} | ${slideNum} | ${title} |`);
    });

} catch (e) {
    console.error("Error:", e.message);
}

File: analyze_slides.py

import re
import sys

def analyze_presentation(file_path):
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
    except Exception as e:
        print(f"Error reading file: {e}")
        return

    # Split by slide div to be approximate but robust enough
    # Assuming <div class="slide ..."> or <div class="slide">
    
    # Find all slide blocks
    # We can use regex to find start indices of slides
    slide_starts = [m.start() for m in re.finditer(r'<div class="slide', content)]
    
    slides = []
    for i in range(len(slide_starts)):
        start = slide_starts[i]
        end = slide_starts[i+1] if i + 1 < len(slide_starts) else len(content)
        slides.append(content[start:end])

    print(f"# Presentation Analysis Report")
    print(f"**File:** `{file_path}`")
    print(f"**Total Slides:** {len(slides)}")
    print("\n## Slide Overview")
    print("| Index | Slide Number | Title |")
    print("| :--- | :--- | :--- |")

    for i, slide_content in enumerate(slides):
        title = "No Title"
        slide_num = "-"

        # Extract Header H1
        # Look for <header>...<h1>Title</h1>...</header> or just <h1> inside the slide
        
        # Simple regex for h1
        h1_match = re.search(r'<h1[^>]*>(.*?)</h1>', slide_content, re.DOTALL | re.IGNORECASE)
        if h1_match:
            raw_title = h1_match.group(1)
            # Remove tags from title if any
            clean_title = re.sub(r'<[^>]+>', '', raw_title)
            title = clean_title.strip()
            
            # extract number
            num_match = re.match(r'(\d+)\.', title)
            if num_match:
                slide_num = num_match.group(1)

        # Check footer for page number
        # <div class="footer">...<span>Page 01</span></div>
        footer_match = re.search(r'class="footer"[^>]*>.*?<span>(.*?)</span>\s*</div>', slide_content, re.DOTALL | re.IGNORECASE)
        if footer_match:
             # The footer usually has two spans, we want the last one possibly?
             # My regex above grabs the content of the *last* span if it's greedy? No, non-greedy.
             # Let's simple check spans inside footer
             
             footer_content = footer_match.group(0)
             spans = re.findall(r'<span>(.*?)</span>', footer_content, re.DOTALL)
             if spans:
                 last_span = spans[-1]
                 if 'Page' in last_span:
                     slide_num = last_span.replace('Page', '').strip()
                 elif last_span.strip().isdigit():
                     slide_num = last_span.strip()

        # Clean title
        title = title.replace('\n', ' ').replace('\r', '')
        # Remove extra spaces
        title = re.sub(r'\s+', ' ', title).strip()
        
        if len(title) > 60:
            title = title[:57] + "..."
            
        print(f"| {i} | {slide_num} | {title} |")

if __name__ == "__main__":
    analyze_presentation(r"c:\Users\ODN\.gemini\antigravity\scratch\isedss-presentation-new\presentation.html")
